.github/workflows/rust-ci.yaml

```diff
@@ -0,0 +1,53 @@
+name: Rust CI
+
+on:
+  merge_group:
+  pull_request:
+  push:
+    # We need to explicitly include tags because otherwise when adding
+    # `branches-ignore` it will only trigger on branches.
+    tags:
+      - '*'
+    branches-ignore:
+      # Ignore pushes to merge queues.
+      # We only want to test the merge commit (`merge_group` event), the hashes
+      # in the push were already tested by the PR checks
+      - 'gh-readonly-queue/**'
+      - 'dependabot/**'
+  workflow_dispatch:
+
+jobs:
+  test:
+    runs-on: ubuntu-22.04
+
+    steps:
+      - name: Fetch sources
+        uses: actions/checkout@v4
+        with:
+          submodules: recursive
+
+      - name: Run tests
+        uses: frequenz-floss/gh-action-cargo-test@v1.0.0
+
+  release:
+    runs-on: ubuntu-latest
```

Comment:
And here. Maybe it would be best to be more consistent. Ideally the latest Ubuntu version but with the version pinned (like `ubuntu-24.4`). But it is not always possible, sometimes an older version is needed to make sure it works with older controllers.